### PR TITLE
♻️ redirect `list_dynamic_services` via dynamic-scheduler

### PR DESCRIPTION
**Pull request by GitHK**
## What do these changes do?

Redirecting `list_dynamic_services` via dynamic-scheduler. Webserver calls the endpoint via RPC like it does for: starting, stopping and getting the status of a service.

**NOTE:** this is very noisy due to the mocks rename to better reflect the new circumstances.

## Related issue/s

- https://github.com/ITISFoundation/osparc-simcore/issues/5632
- https://github.com/ITISFoundation/osparc-simcore/issues/6894
- closes https://github.com/ITISFoundation/osparc-simcore/issues/6895
- should fix https://github.com/ITISFoundation/osparc-simcore/issues/6774

## Dev-ops checklist

- [x] No ENV changes or I properly updated ENV ([read the instruction](https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/blob/configs/README.md?ref_type=heads#how-to-update-env-variables))

### CODE REVIEW

**Review by sanderegg — APPROVED** [review]
thanks

**Review by matusdrobuliak66 — APPROVED** [review]
Thanks!

File: services/dynamic-scheduler/src/simcore_service_dynamic_scheduler/services/director_v2/_thin_client.py [hunk 1]
```diff
@@ -108,3 +110,15 @@ async def _(
             )
 
         return await _(self)
+
+    @retry_on_errors()
+    @expect_status(status.HTTP_200_OK)
+    async def get_dynamic_services(
+        self, *, user_id: UserID | None = None, project_id: ProjectID | None = None
```
Comment (pcrespov) [inline services/dynamic-scheduler/src/simcore_service_dynamic_scheduler/services/director_v2/_thin_client.py:117]: TIP: consider using this technique with `unset` sentinels and the helper function to build `params`
https://github.com/itisfoundation/osparc-simcore/blob/59aeb9964941234c040ac1c39eec9a8217db244d/services/web/server/src/simcore_service_webserver/folders/_folders_db.py#L314-L333

NOTE: if you use it , please move it from `utils` to `common_library`

File: services/web/server/src/simcore_service_webserver/projects/_trash_api.py [hunk 2]
```diff
@@ -56,8 +57,8 @@ async def _is_project_running(
             app, user_id=user_id, project_id=project_id
         )
     ) or bool(
-        await director_v2_api.list_dynamic_services(
-            app, user_id=user_id, project_id=f"{project_id}"
+        await dynamic_scheduler_api.list_dynamic_services(
```
Comment (pcrespov) [inline services/web/server/src/simcore_service_webserver/projects/_trash_api.py:60]: Can `list_dynamic_services` raise differently to the old one? i.e. can you document a bit in the function some known exceptions?

File: services/web/server/src/simcore_service_webserver/dynamic_scheduler/api.py [hunk 3]
```diff
@@ -18,19 +18,30 @@
...
 from ..rabbitmq import get_rabbitmq_client, get_rabbitmq_rpc_client
 from .settings import DynamicSchedulerSettings, get_plugin_settings
 
 _logger = logging.getLogger(__name__)
 
 
+async def list_dynamic_services(
+    app: web.Application,
+    *,
+    user_id: UserID | None = None,
```
Comment (pcrespov) [inline services/web/server/src/simcore_service_webserver/dynamic_scheduler/api.py:37]: Q: what do these calls return? It is a list of dynamic services in ... 

-1.  `await list_dynamic_services(app)` ??
-2.  `await list_dynamic_services(app, user_id=1)` ??
-3.  `await list_dynamic_services(app, project_id="(some uuid)")` ??
-4.  `await list_dynamic_services(app, user_id=1, project_id="(some uuid)")` ??


Can you also document "known" exceptions if possible?

Comment (GitHK) [reply services/web/server/src/simcore_service_webserver/dynamic_scheduler/api.py:37]: the current running dynamic services. Let's add a small docstring and not keep it as the original.

Comment (GitHK) [reply services/web/server/src/simcore_service_webserver/dynamic_scheduler/api.py:37]: There are no known exceptions

File: services/web/server/src/simcore_service_webserver/projects/_trash_api.py (hunk 2 above)
Comment (GitHK) [reply services/web/server/src/simcore_service_webserver/projects/_trash_api.py:60]: It raises different errors for sure. But does not raise any extra known errors.

File: services/web/server/src/simcore_service_webserver/folders/_folders_db.py [hunk 4]
```diff
@@ -312,12 +312,12 @@ async def update(
     folders_id_or_ids: FolderID | set[FolderID],
     product_name: ProductName,
     # updatable columns
-    name: str | UnSet = _unset,
-    parent_folder_id: FolderID | None | UnSet = _unset,
-    trashed_at: datetime | None | UnSet = _unset,
-    trashed_explicitly: bool | UnSet = _unset,
-    workspace_id: WorkspaceID | None | UnSet = _unset,
-    user_id: UserID | None | UnSet = _unset,
+    name: str | UnSet = UnSet.INSTANCE,
```
Comment (pcrespov) [inline services/web/server/src/simcore_service_webserver/folders/_folders_db.py:315]: I liked more how it looked before but ok

File: packages/common-library/src/common_library/unset.py [hunk 5]
```diff
@@ -0,0 +1,12 @@
+from typing import Any
+
+
+class UnSet:
+    INSTANCE: "UnSet"
+
+
+UnSet.INSTANCE = UnSet()
```
Comment (pcrespov) [inline packages/common-library/src/common_library/unset.py:8]: I like more a simple `unset`  object.  At the end of the day the helper below looks for a instance of UnSet. I do not wee why we have to create a unique instance for it. :-)

File: services/web/server/src/simcore_service_webserver/dynamic_scheduler/api.py (hunk 3 above)
Comment (pcrespov) [reply services/web/server/src/simcore_service_webserver/dynamic_scheduler/api.py:37]: What about the question above? what is the expected result on each of the 4 cases above

**Review by pcrespov — APPROVED** [review]
OK. Please check the questions i left. thx

File: services/web/server/src/simcore_service_webserver/dynamic_scheduler/api.py (hunk 3 above)
Comment (GitHK) [reply services/web/server/src/simcore_service_webserver/dynamic_scheduler/api.py:37]: So to answer you questions it's always a list of dynamic services, both legacy and new style filtered by project_id and user_id independently. You can mix them as you please. All above options are accepted.
We generally use options: `1` and `4`. The other two are also supported

**Review by giancarloromeo — APPROVED** [review]
looks good

**Pull request merged**